File: src/app/services/nonstp-reason.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class NonstpReasonService {
  private apiUrl = 'http://localhost:8080/api/nonstp-reason';

  constructor(private http: HttpClient) {}

  clearNonstpReason(): void {
    // Logic to clear the non-stop reason and set the recheck and stop flags to 'Y'
    this.setFlags();
  }

  setFlags(): void {
    // Logic to set the recheck and stop flags to 'Y'
    // Assuming we have some flags in the service to set
    this.recheckFlag = 'Y';
    this.stopFlag = 'Y';
  }

  saveNonstpReason(nonstpReason: string): Observable<any> {
    // Make an HTTP POST request to the back-end API to save the non-stop reason
    return this.http.post(`${this.apiUrl}/save`, { nonstpReason });
  }

  updateScrutinyPartnerForm(applicationNumber: string, previousPolicyDetails: any): Observable<any> {
    // Make an HTTP POST request to the back-end API to update the scrutiny partner form with the provided details
    return this.http.post(`${this.apiUrl}/update-scrutiny-partner-form`, { applicationNumber, previousPolicyDetails });
  }
}

File: src/main/java/com/azbj/proposalde3/service/NonstpReasonService.java
package com.azbj.proposalde3.service;

import com.azbj.proposalde3.repository.NonstpReasonRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class NonstpReasonService {

    @Autowired
    private NonstpReasonRepository nonstpReasonRepository;

    public void clearNonstpReason() {
        // Clear the non-stop reason and set the recheck and stop flags to 'Y'
        nonstpReasonRepository.updateNonstpReason("", 'Y', 'Y');
    }

    public void saveNonstpReason(String nonstpReason) {
        // Save the non-stop reason in the database
        nonstpReasonRepository.save(nonstpReason);
    }

    public void updateScrutinyPartnerForm(String applicationNumber, Object previousPolicyDetails) {
        // Update the scrutiny partner form with the provided details in the database
        nonstpReasonRepository.updateScrutinyPartnerForm(applicationNumber, previousPolicyDetails);
    }
}
